pandas/core/groupby/generic.py

```diff
@@ -1107,21 +1103,35 @@ def cast_result_block(result, block: "Block", how: str) -> "Block":
                         # is a lie. To keep the code-path for the typical non-split case
                         # clean, we choose to clean up this mess later on.
                         assert len(locs) == result.shape[1]
+                        new_blocks = []
```

Comment:
can you type this

Reply:
can you type this

Reply:
and move this to a higher scope (as used in both parts)

Reply:
We only append to it in this scope, everywhere else we set it directly

Reply:
this is used in 3 blackes (sure you don't append and instead directly assign in the other scopes), but it is the same name. I would simply define it once, then append for each (or leave the assign for the others is ok i guess); it just is much easier to grok the fact that we are defining the same variable.

Reply:
done+green